src/query/expression/src/types/number.rs

```diff
@@ -549,24 +549,29 @@ impl<T: Number> SimpleDomain<T> {
     }
 
     pub fn overflow_cast_with_minmax<U: Number>(&self, min: U, max: U) -> (SimpleDomain<U>, bool) {
-        let (min, min_overflowing) = overflow_cast_with_minmax::<T, U>(self.min, min, max);
-        let (max, max_overflowing) = overflow_cast_with_minmax::<T, U>(self.max, min, max);
+        let (min, min_overflowing) = overflow_cast_with_minmax::<T, U>(self.min, min, max, min);
+        let (max, max_overflowing) = overflow_cast_with_minmax::<T, U>(self.max, min, max, max);
+
         (
             SimpleDomain { min, max },
             min_overflowing || max_overflowing,
         )
     }
 }
 
-fn overflow_cast_with_minmax<T: Number, U: Number>(src: T, min: U, max: U) -> (U, bool) {
+fn overflow_cast_with_minmax<T: Number, U: Number>(
+    src: T,
+    min: U,
+    max: U,
+    fallback: U,
+) -> (U, bool) {
```

Comment:
```suggestion
fn overflow_cast_with_minmax<T: Number, U: Number>(
    src: T,
    min: U,
    max: U,
) -> Option<(U, bool)> {
```

Reply:
For float to int, there's nan/inf case that we can't use unwrap.